### ISSUE TITLE
When I set the operator weights all to 1, the operator weights change as soon as an input is made to the operator

### ISSUE THREAD

**Issue by CZXIANGOvO** [pytorch/pytorch#124998]
### 🐛 Describe the bug

When I set the operator weights all to 1, the operator weights change as soon as an input is made to the operator.Before and after taking input to the operator, we output its state_dict(), and found that the two are different.

```python
import torch
import time
import numpy as np

if __name__ == "__main__":
    model_path = '/root/czx/log/batchnorm2d.pth'
    model = torch.load(model_path)
    print(model.state_dict())
    tensor = torch.rand(5,64,224,224).to('cuda:0')
    tensor = tensor.clamp(min=0, max=1)
    model(tensor)
    print(model.state_dict())
```
[batchnorm2d.zip](https://github.com/pytorch/pytorch/files/15124733/batchnorm2d.zip)

### Versions

ubuntu 18.04
torch-GPU

**Comment by mikaylagawarecki** [comment]
Could you share the output please?

**Comment by CZXIANGOvO** [comment]
> Could you share the output please?
The top dict is for weights that are all 1, and the bottom dict is for weights that have changed after input. If the input is large, the weights will change even more.
![屏幕截图 2024-04-25 210127](https://github.com/pytorch/pytorch/assets/121651120/3f86728b-1c6d-4cf0-b945-a337d9d2eb0e)

**Comment by CZXIANGOvO** [comment]
Currently I'm testing the batchnorm2d operator in both mindspore and tensorflow frameworks and none of the weights change.
```python
import torch
import numpy as np
from mindspore import dtype as mstype
from mindspore import nn, Tensor
import mindspore
from tensorflow import nn as tf_nn
import tensorflow as tf
import keras
def torch_bn2():
    model_path = './batchnorm2d.pth'
    model = torch.load(model_path)
    print("**************torch-before-state_dict()*****************")
    print(model.state_dict())
    tensor = torch.rand(5, 64, 224, 224).to('cuda:0')
    tensor = tensor.clamp(min=0, max=1)
    model(tensor)
    print("**************torch-after-state_dict()*****************")
    print(model.state_dict())



def mindspore_bn2():
    network = nn.BatchNorm2d(100)
    param = network.parameters_dict()["beta"]

    update = nn.ParameterUpdate(param)
    weight = Tensor(np.ones(100), mstype.float32)
    output = update(weight)
    param = network.parameters_dict()["beta"]
    print("**************mindspore-before-state_dict()*****************")
    print("Parameter:\n", param.asnumpy())
    input_x = Tensor(np.ones([1, 100, 100, 100]), mindspore.float32)
    output = network(input_x)
    param = network.parameters_dict()["beta"]
    print("**************mindspore-after-state_dict()*****************")
    print("Parameter:\n", param.asnumpy())


def tensorflow_bn2():

    model = keras.models.Sequential([
        keras.layers.Conv2D(32, (3, 3), activation='relu', input_shape=(28, 28, 1)),
        keras.layers.BatchNormalization(),  
        keras.layers.MaxPooling2D((2, 2)),
        keras.layers.Conv2D(64, (3, 3), activation='relu'),
        keras.layers.BatchNormalization(), 
        keras.layers.MaxPooling2D((2, 2)),
        keras.layers.Conv2D(64, (3, 3), activation='relu')
    ])


    t = model.layers[1].get_weights()

    zeros_list = np.zeros(32)
    t[0] = zeros_list
    model.layers[1].set_weights(t)
    print("**************tensorflow-before-state_dict()*****************")
    print(model.layers[1].get_weights())
    print(type(model.layers[1]))
    input_data = tf.random.normal([32, 28, 28, 1])

    output_data = model(input_data)
    print("**************tensorflow-after-state_dict()*****************")
    print(model.layers[1].get_weights())
    print(type(model.layers[1]))

if __name__ == "__main__":
    torch_bn2()
    mindspore_bn2()
    tensorflow_bn2()
```
torch
![image](https://github.com/pytorch/pytorch/assets/121651120/4430cc15-81e9-40bb-b497-068fcf7013e4)

mindspore
![image](https://github.com/pytorch/pytorch/assets/121651120/a2113915-2f2e-4d62-9547-eb6c2bf4c3cf)

tensorflow
![image](https://github.com/pytorch/pytorch/assets/121651120/1fb4f921-d4ff-4ced-bc9e-00b1e95fa12c)
